docprompt/provenance/search.py

```diff
@@ -40,24 +40,23 @@ class DocumentProvenanceLocator:
 
     @classmethod
     def from_document_node(cls, document_node: "DocumentNode"):
-        # TODO: See if we can remove the ocr_results attribute from the
-        # PageNode and just use the metadata.task_result["<provider>_ocr"],
-        # result of the OCR task instead.
-
         index = create_tantivy_document_wise_block_index()
         block_mapping_dict = {}
         geo_index_dict: DocumentProvenanceGeoMap = {}
 
         writer = index.writer()
 
         for page_node in document_node.page_nodes:
-            if (
-                not page_node.ocr_results.result
-                or not page_node.ocr_results.result.block_level_blocks
-            ):
+            task_keys = page_node.metadata.task_results.keys()
+            ocr_keys = [key for key in task_keys if key.endswith("_ocr")]
+
+            if not ocr_keys:
                 continue
 
-            ocr_result = page_node.ocr_results.result
+            results = [page_node.metadata.task_results[key] for key in ocr_keys]
+
+            # Get the result with the most recent `when` attribute
+            ocr_result = sorted(results, key=lambda x: x.when)[0]
```

Comment:
Is when guaranteed to be a comparable?

Reply:
When is timestamp created by the result model through a default factory